delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/IdSegmentRef.scala

```diff
@@ -0,0 +1,61 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.model
+
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+
+/**
+  * A segment from the positional API that should be an Id plus its revision/tag information
+  */
+sealed trait IdSegmentRef extends Product with Serializable { self =>
```

Comment:
We have now 2 ADTs describing the Latest/Tag/Rev where we will have a 4th one to get a value by timestamp.
It may be possible to have only one ADT to describe the 4 of them no ?

Reply:
What 2 ADTs? Do you mean `ResourceRef`? And there is nothing YET about getting a value by timestamp. I don't see the point of thinking ahead here.

Reply:
To avoid to refactor most of this again ?

Reply:
Things are to be refactored when the time is needed, not prematurely.
We can try to consolidate `ResourceRef` and `IdSegmentRef` in one single thing, but it is not going to be as trivial. This PR is already quite big, so I suggest we do that in another one if we see it useful.

Reply:
We might consider that later on.